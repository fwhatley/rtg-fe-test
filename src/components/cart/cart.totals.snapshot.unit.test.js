import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import CartTotals from './cart.totals';
import {
    getProductListWithQuantity,
    mockStore,
} from '../../test.utils/test.utils';

it('displays CartTotals with no products', () => {
    const store = mockStore({
        cart: {
            items: [],
        },
    });

    const tree = renderer.create(
        <Provider store={store}>
            <CartTotals />
        </Provider>
    );
    expect(tree).toMatchSnapshot();
});

it('displays CartTotals with some products', () => {
    const store = mockStore({
        cart: {
            items: getProductListWithQuantity(),
        },
    });

    const tree = renderer.create(
        <Provider store={store}>
            <CartTotals />
        </Provider>
    );
    expect(tree).toMatchSnapshot();
});
