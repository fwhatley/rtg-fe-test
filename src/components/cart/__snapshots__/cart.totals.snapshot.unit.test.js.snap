// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays CartTotals with no products 1`] = `
<div
  className="cell small-12"
  style={
    Object {
      "paddingTop": 50,
    }
  }
>
  <div
    className="grid-x"
  >
    <div
      className="cell small-3"
    >
      SUBTOTAL
    </div>
    <div
      className="cell small-3"
    >
      $ 
      0.00
    </div>
  </div>
  <div
    className="grid-x"
  >
    <div
      className="cell small-3"
    >
      TAX @10%
    </div>
    <div
      className="cell small-3"
    >
      $ 
      0.00
    </div>
  </div>
  <div
    className="grid-x"
  >
    <div
      className="cell small-3"
    >
      SHIPPING COST
    </div>
    <div
      className="cell small-3"
    >
      $ 
      50.00
    </div>
  </div>
  <div
    className="grid-x"
  >
    <div
      className="cell small-3"
    >
      <strong>
        TOTAL
      </strong>
    </div>
    <div
      className="cell small-3"
    >
      <strong>
        $ 
        50.00
      </strong>
    </div>
  </div>
</div>
`;

exports[`displays CartTotals with some products 1`] = `
<div
  className="cell small-12"
  style={
    Object {
      "paddingTop": 50,
    }
  }
>
  <div
    className="grid-x"
  >
    <div
      className="cell small-3"
    >
      SUBTOTAL
    </div>
    <div
      className="cell small-3"
    >
      $ 
      9164.94
    </div>
  </div>
  <div
    className="grid-x"
  >
    <div
      className="cell small-3"
    >
      TAX @10%
    </div>
    <div
      className="cell small-3"
    >
      $ 
      916.49
    </div>
  </div>
  <div
    className="grid-x"
  >
    <div
      className="cell small-3"
    >
      SHIPPING COST
    </div>
    <div
      className="cell small-3"
    >
      $ 
      188.30
    </div>
  </div>
  <div
    className="grid-x"
  >
    <div
      className="cell small-3"
    >
      <strong>
        TOTAL
      </strong>
    </div>
    <div
      className="cell small-3"
    >
      <strong>
        $ 
        10269.73
      </strong>
    </div>
  </div>
</div>
`;
